nion/swift/ImageCanvasItem.py

```diff
@@ -255,34 +263,78 @@ def _repaint(self, drawing_context: DrawingContext.DrawingContext, canvas_bounds
         scale_marker_height = 6
         scale_marker_font = "normal 14px serif"
         get_font_metrics_fn = self.__get_font_metrics_fn
-        if dimensional_calibration:  # display scale marker?
+
+        if dimensional_calibration and screen_pixel_per_image_pixel and screen_pixel_per_image_pixel > 0.0:
+            scale_marker_image_width = scale_marker_width / screen_pixel_per_image_pixel
+            calibrated_scale_marker_width = Geometry.make_pretty2(scale_marker_image_width * dimensional_calibration.scale, True)
+            scale_marker_image_width = calibrated_scale_marker_width / dimensional_calibration.scale
+            scale_marker_calculated_width = scale_marker_image_width * screen_pixel_per_image_pixel
+
```

Comment:
Formatting issue (should be one blank line). Also check that uses of `str | None` have spaces around the ` | ` character.